src/features/logging/aggregate/index.js

```diff
@@ -10,21 +10,43 @@ import { Log } from '../shared/log'
 import { isValidLogLevel } from '../shared/utils'
 import { applyFnToProps } from '../../../common/util/traverse'
 import { MAX_PAYLOAD_SIZE } from '../../../common/constants/agent-constants'
-import { FEATURE_TO_ENDPOINT } from '../../../loaders/features/features'
+import { FEATURE_NAMES, FEATURE_TO_ENDPOINT } from '../../../loaders/features/features'
+import { SESSION_EVENT_TYPES, SESSION_EVENTS } from '../../../common/session/constants'
+import { ABORT_REASONS } from '../../session_replay/constants'
+import { canEnableSessionTracking } from '../../utils/feature-gates'
 
 export class Aggregate extends AggregateBase {
   static featureName = FEATURE_NAME
   constructor (agentRef) {
     super(agentRef, FEATURE_NAME)
     this.harvestTimeSeconds = agentRef.init.logging.harvestTimeSeconds
 
+    // The SessionEntity class can emit a message indicating the session was cleared and reset (expiry, inactivity). This feature must abort and never resume if that occurs.
+    this.ee.on(SESSION_EVENTS.RESET, () => {
+      this.abort(ABORT_REASONS.RESET)
+    })
+
+    this.ee.on(SESSION_EVENTS.UPDATE, (type, data) => {
+      if (this.blocked || type !== SESSION_EVENT_TYPES.CROSS_TAB) return
+      if (this.mode !== LOGGING_MODE.OFF && data.loggingMode === LOGGING_MODE.OFF) this.abort(ABORT_REASONS.CROSS_TAB)
+      else this.mode = data.loggingMode
+    })
+
     this.waitForFlags(['log']).then(([loggingMode]) => {
-      if (!loggingMode) {
+      const session = this.agentRef.runtime.session
+      if (this.loggingMode === LOGGING_MODE.OFF || (session.isNew && loggingMode === LOGGING_MODE.OFF)) {
         this.blocked = true
         this.deregisterDrain()
         return
       }
-      this.loggingMode = loggingMode
+      if (session.isNew || !canEnableSessionTracking(this.agentRef.agentIdentifier)) {
+        this.loggingMode = loggingMode
+        this.syncWithSessionManager({
+          loggingMode: this.loggingMode
+        })
```

Comment:
I think we need to double check that this wont cause issues. One of the two cases that could trigger this line is that session track is disabled, meaning there is no session object.  I fear calling this method could cause errors when that is the case

Reply:
Should probably only try to sync with the session manager if we know it exists for sure. Either handled here or in the base class method